### PR TITLE
[coq] [doc] Some fixes to documentation.

### PR DESCRIPTION
**Pull request by ejgallego**
In particular the advice about directory split was bogus.

### CODE REVIEW

**Comment by Alizter** [comment]
The directory split advice was the workaround to the META issue that has now been fixed. So it wasn't bogus at some point.

File: doc/coq.rst [hunk 1]
```diff
@@ -16,8 +16,13 @@ and ``.mlg`` file preprocessing.
 A *Coq theory* is a collection of ``.v`` files that define Coq modules whose
 names share a common prefix. The module names reflect the directory hierarchy.
 
-A *Coq plugin* is an OCaml :ref:`library` that Coq can load dynamically at
-runtime. Plugins are typically linked with the Coq OCaml API.
+A *Coq plugin* is an OCaml :ref:`library` that Coq can load
```
Comment (Alizter) [inline doc/coq.rst:19]: Why did you wrap this tighter?

**Comment by ejgallego** [comment]
> The directory split advice was the workaround to the META issue that has now been fixed. So it wasn't bogus at some point.

I think it was bogus back then too.

The META issue for me == a missing dependency ; what do you mean by "META issue"?

How does splitting into two directories help for a missing dependency to appear?

File: doc/coq.rst (hunk 1 above)
Comment (ejgallego) [reply doc/coq.rst:19]: Emacs decided to do it.

File: doc/coq.rst [hunk 2]
```diff
@@ -104,9 +110,11 @@ The semantics of the fields are:
   theory, which means the ``Coq`` prefix won't be bound, and ``Coq.Init.Prelude``
   won't be imported by default.
 
-- The path to the installed locations of the ``<ocaml_plugins>`` is passed to
-  ``coqdep`` and ``coqc`` using Coq's ``-I`` flag. This allows a Coq theory to
-  depend on OCaml plugins.
+- If ``plugins`` is present, Dune will pass the corresponding flags to
```
Comment (Alizter) [inline doc/coq.rst:113]:
```suggestion
- If the ``plugins`` field is present, Dune will pass the corresponding flags to
```

**Comment by ejgallego** [comment]
Note that #6167 still solves nothing, `coqdep` (and likely `coqc`) are still very much broken for any build setup which is not the one used in Coq's test suite.

File: doc/coq.rst [hunk 3]
```diff
@@ -104,9 +110,11 @@ The semantics of the fields are:
...
+  Coq so that ``coqdep`` and ``coqc`` can find the corresponding OCaml
+  libraries declared in ``<ocaml_plugins>``. This allows a Coq theory
+  to depend on OCaml plugins. Starting with ``(lang coq 0.6)``,
```
Comment (Alizter) [inline doc/coq.rst:116]: I thought we checked for private plugins before 0.6? I may be confused however. I seem to remember introducing the limitation in the workspace composition PR.

File: doc/coq.rst (hunk 2 above)
Comment (ejgallego) [reply doc/coq.rst:113]: Note both styles appear in the doc, we should pick one. "If ``package`` is present ..."

**Comment by Alizter** [comment]
> > The directory split advice was the workaround to the META issue that has now been fixed. So it wasn't bogus at some point.
> 
> I think it was bogus back then too.
> 
> The META issue for me == a missing dependency ; what do you mean by "META issue"?
> 
> How does splitting into two directories help for a missing dependency to appear?

It caused the META file to be build in time by chance. It was only a workaround.

File: doc/coq.rst (hunk 3 above)
Comment (ejgallego) [reply doc/coq.rst:116]: I think that was only checked if the plugins live in a different scope.

With the legacy loading method it is fine to have private libs as plugins, only now that we depend (unconditionally) on the META file this is a hard limitation.

File: doc/coq.rst (hunk 1 above)
Comment (Alizter) [reply doc/coq.rst:19]: Can you make it consistent with the rest of the file? Emacs should let you modify the fill-column variable to 80.

**Comment by ejgallego** [comment]
> It caused the META file to be build in time by chance. It was only a workaround.

How did having two directories cause the META file to build?

File: doc/coq.rst (hunk 3 above)
Comment (Alizter) [reply doc/coq.rst:116]: Private plugins don't make any sense currently in Coq (only for private theories). If you have a public theory then you can install .vo files, however they will rely on the plugin to be installed somewhere to work hence it doesn't make sense to support this.

**Comment by Alizter** [comment]
@ejgallego No idea, that was how we worked around the issue.

File: doc/coq.rst (hunk 3 above)
Comment (ejgallego) [reply doc/coq.rst:116]: Not sure I understand what you mean. Indeed it is logical that private plugins do only make sense for private theories, but that's a valid use case we could support.

**Comment by ejgallego** [comment]
> @ejgallego No idea, that was how we worked around the issue.

You think you workarounded the issue but I bet you did not, because for Dune that split makes 0 difference, unless I'm missing something. You were likely confused by the terrible (non deterministic) coqdep code.

Maybe a problem was with `-I ` pointing out to a place with a stale META?

But if so, that problem still persists no matter what you do with the dirs. Needs to be fixed at the coq lang version level.

File: doc/coq.rst (hunk 3 above)
Comment (Alizter) [reply doc/coq.rst]: Right I was talking about before. Having private libs in private theories was the only use case that made sense. Private libs in public theories doesn't make any sense. Now with the new loading method however, it means that even private theories can only import public libs which is not currently restricted by Dune IIRC.

To allow for the user case you mention, we would need to yet again change how Coq loads plugins.

Comment (ejgallego) [reply doc/coq.rst]: We don't need to change how Coq loads plugins, it is enough for Dune to setup a virtual findlib env where the private libraries are exposed. But that's quite some work tho.

File: doc/coq.rst [hunk 4]
```diff
@@ -592,9 +597,9 @@ Together with ``hello_world.ml``:
   let hello_world = "hello world!"
 
 They make up the plugin. There is one more important ingredient here, and that
```
Comment (christinerose) [inline doc/coq.rst:599]:
```suggestion
They make up the plugin. There is one more important ingredient here and that
```

**Review by christinerose — APPROVED** [review]
Looks good! I just found two commas I'd suggest removing.

**Comment by ejgallego** [comment]
Thanks @christinerose !

**Pull request merged**